inference-engine/src/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/compress_float_constants.hpp"
+
+#include "openvino/opsets/opset8.hpp"
+#include "ngraph/rt_info.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/rt_info/decompression.hpp"
+#include "transformations/rt_info/disable_fp16_compression.hpp"
+#include "transformations/rt_info/old_api_map_attribute.hpp"
+#include "itt.hpp"
+
+
+namespace {
+template <ov::element::Type_t PREC_FROM, ov::element::Type_t PREC_TO>
+std::shared_ptr<ov::Node> change_constant_precision(std::shared_ptr<ov::opset8::Constant>& constant) {
+    using src_type = typename ov::element_type_traits<PREC_FROM>::value_type;
+    using dst_type = typename ov::element_type_traits<PREC_TO>::value_type;
+
+    const auto* src_data = constant->get_data_ptr<src_type>();
+    const auto size = ov::shape_size(constant->get_shape());
+
+    auto new_constant = std::make_shared<ov::opset8::Constant>(PREC_TO, constant->get_shape());
+    auto* dst_data = const_cast<dst_type*>(reinterpret_cast<const dst_type*>(new_constant->get_data_ptr()));
+    if (dst_data == nullptr)
+        return nullptr;
+
+    for (size_t i = 0; i < size; ++i) {
+        if (src_data[i] > std::numeric_limits<dst_type>::max()) {
+            dst_data[i] = std::numeric_limits<dst_type>::max();
+        } else if (src_data[i] < std::numeric_limits<dst_type>::lowest()) {
+            dst_data[i] = std::numeric_limits<dst_type>::lowest();
```

Comment:
shouldn't we print log message about constant value pruning? In this case a model gets potentially incorrectly inferable.